core/unit_test/TestLocalDeepCopy.hpp

```diff
@@ -333,9 +333,14 @@ void impl_test_local_deepcopy_teampolicy_rank_5(const int N) {
   using member_type = typename Kokkos::TeamPolicy<ExecSpace>::member_type;
 
   // Deep Copy
+  team_policy policy(N, Kokkos::AUTO);
+  // FIXME_SYCL improve default team_size
+#if defined(KOKKOS_ENABLE_SYCL) && !defined(KOKKOS_ARCH_INTEL_GPU)
+  if (std::is_same<ExecSpace, Kokkos::Experimental::SYCL>::value)
+    policy = team_policy(N, 512);
+#endif
```

Comment:
Yes, I think I'm requesting too many registers here on V100 but it seems to work fine on `Intel GPUs`...

Reply:
The calculations for default/maximum team size don't take registers into account (since this information is in general not available through the `SYCL` interface). There are multiple places where we need to use a smaller workgroup size for the SYCL+CUDA CI (where it's not an issue for Intel GPUs).

Reply:
shouldn't we then be generally more careful with the max size return? I.e. limit ourself to 256 threads if building for NVIDIA GPUs for example? That would mean that it will always fit.

Reply:
OK, I'll try that.